src/ShipAICmd.cpp

```diff
@@ -434,12 +447,12 @@ bool AICmdKill::TimeStepUpdate()
 		double vissize = 1.3 * m_dBody->GetPhysRadius() / dist;
 		vissize += (0.05 + 0.5 * leaddiff) * Pi::rng.Double() * skillShoot;
 		if (vissize > headdiff)
-			m_fguns->SetGunFiringState(0, 1);
+			m_guns->SetAllGroupsFiring(true);
 		else
-			m_fguns->SetGunFiringState(0, 0);
-		float max_fire_dist = m_fguns->GetGunRange(0);
+			m_guns->SetAllGroupsFiring(false);
+		float max_fire_dist = stats.range;
 		if (max_fire_dist > 4000) max_fire_dist = 4000;
```

Comment:
```suggestion
		float max_fire_dist = std::min(stats.range, 4000.0f);
```
(It took me a few seconds to figure out why the local variable was needed in the first place; this makes it marginally more obvious IMHO)

I know this is not part of this PR, but why are we clamping a weapon range to some arbitrary maximum range here instead of just trusting the weapon stats?

Reply:
Ah, fair, it's just for AI ships? (I should probably look at the filename lol)  So player-ship-weapons are not affected by this clamping; gotcha.

Interesting, so flying 4km's from a target and taking pot-shots at distance guarantees a safe fight..  this requires further testing!  (Perhaps it was deemed too difficult/unfair for AI ships to have perfect aim from so far away..) Then again, AI should have a skill modifier, which adds a skill-scaled amount of jitter to the aiming. (No idea whether or not it does, just saying)